content/docs/custom-plugins/index.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Installing Custom Plugins
+lastUpdated: '2021-03-16T17:55:00.0Z'
+description: How to add a custom Backstage plugin to Roadie
+---
+
+# Introduction
+
+Your plugins can be installed via npm or yarn, like publishing a normal package. We will provide you a unique scope for
+your plugins, and we host a dedicated npm registry which you can publish to.
+
+## Prerequisites
+
+- For our alpha release you'll need to give us some information upfront about your plugin. We plan to make all of this
+  self-service going forward but currently backstage doesn't support this out-of-the-box. For now, we will need you to
+  fill out [this form][form].
+
+- Get your credentials for the npm registry. Follow [this link][forgot-password] to reset your password. The username is
+  `<your-company>-roadie`. This will send a reset link to the email we have associated with your account. You can
+  change this email as you wish.
+
+- Configure npm. There are two options:
+
+  - Interactively using npm login. Suitable for a developer testing the process.
+    <pre>
+    npm config set @<b>&lt;your-company&gt;</b>-roadie:registry https://roadiehq.jfrog.io/artifactory/api/npm/<b>&lt;your-company&gt;</b>-roadie/
+    npm login --scope=@<b>&lt;your-company&gt;</b>-roadie 
+    # This will prompt for your username, password and email
+    </pre>
+
+  - In a CI/CD environment you can write a local `.npmrc` file
+    <pre>
+    @<b>&lt;your-company&gt;</b>-roadie:registry &lt;https://roadiehq.jfrog.io/artifactory/api/npm/<b>&lt;your-company&gt;</b>-roadie/&gt;
+    _auth = username:password # converted to base64
+    # _auth = ${NPM_AUTH} # You can also use an environment variable
+    always-auth = true
+    </pre>
+
+## Publishing
+
+Once you're ready to publish your plugin these are the steps you'll need to follow:
+
+1. Ensure your package is scoped correctly in package.json. The name field should follow this pattern
+   `@<your-company>-roadie/<package-name>`
+2. Select a new package version to publish. You can't overwrite a version once published (e.g. npm version patch)
+3. Build your plugin (e.g. yarn install && yarn tsc && yarn build)
+4. Publish your plugin npm publish
+5. (Optional) Check the registry npm info `@<your-company>-roadie/<package-name>`
+
```

Comment:
Is there a way to verify that everything worked or something we can tell the user to do as a next step? It feels kinda inconclusive down here.

Reply:
AFAIK we don't have anything for that yet. I added this story to the custom plugin epic https://app.clubhouse.io/larder/story/1571/give-user-feedback-on-publishing-operation

This is for the old way of adding a custom plugin so we should decide if we want to build that or just add the feedback in the new way of adding custom plugins. I will ask Iain anyways in case I missed something.

Reply:
Perhaps we could explain how we will notify them when their custom plugin shows up?

It might also be worth saying that future updates to the plugin won't require re-submission of the form or manual steps and thus should be must faster.

Reply:
Not sure what you mean by this
> Perhaps we could explain how we will notify them when their custom plugin shows up?

We can't see when it shows up. Should we say, we will notify them on slack when the first build is ready? 

I will add now the second part you mention

Reply:
Yeah exactly. We haven't told them what happens after they submit the Google form, so they won't know when the plugin is installed and ready to use. We should tell them that we will notify them via their shared Slack channel or via email when the plugin is ready to use after being installed the first time.

Reply:
Added some comments to the prerequisites, and the end of the file. Let me know what you think about those